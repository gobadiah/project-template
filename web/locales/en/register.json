{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
