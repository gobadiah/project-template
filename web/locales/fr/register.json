{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
