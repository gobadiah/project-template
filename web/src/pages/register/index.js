import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
