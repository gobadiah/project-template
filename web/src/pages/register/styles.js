import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
