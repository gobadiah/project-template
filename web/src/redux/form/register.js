import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
