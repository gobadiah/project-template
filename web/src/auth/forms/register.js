import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
