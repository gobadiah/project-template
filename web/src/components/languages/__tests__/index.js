import 'jsdom-global/register';

import React from 'react';
import { shallow } from 'enzyme';

import Languages from '..';
import Lang from '../lang';

const setup = (component, props = {}) => {
  const i18n = require('~/services/i18n').default;
  i18n.changeLanguage = jest.fn();
  i18n.t = jest.fn(key => key);
  i18n.languages = ['fr'];
  const wrapper = shallow(React.createElement(component, props), { context: { i18n, t: i18n.t } });
  return {
    wrapper,
    i18n,
  };
};

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Languages', () => {
  it('should match snapshot', () => {
    const { wrapper } = setup(Languages);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contains two buttons', () => {
    const { wrapper } = setup(Languages);
    expect(wrapper.find(Lang)).toHaveLength(2);
  });

  it('has a method for changing language', () => {
    const { wrapper, i18n } = setup(Languages);
    const lng = 'en';
    wrapper.instance().setLanguage(lng);
    expect(i18n.changeLanguage).toHaveBeenCalledWith(lng);
    expect(window.document.cookie).toEqual(`i18next=${lng}`);
  });

  it('clicking on a button change language', () => {
    const { wrapper, i18n } = setup(Languages);
    expect(i18n.changeLanguage).toHaveBeenCalledTimes(0);
    wrapper.find('button').simulate('click');
    expect(i18n.changeLanguage).toHaveBeenCalledTimes(1);
  });
});

describe('Lang', () => {
  const setupLang = () => setup(Lang, { lng: 'fr', name: 'french', img: 'img' });
  it('should match snapshot', () => {
    const { wrapper } = setupLang();
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain an input', () => {
    const { wrapper } = setupLang();
    expect(wrapper.find('img')).toHaveLength(1);
    /*
    wrapper.find('input').at(0).prop('onClick')();
    expect(i18n.changeLanguage.mock.calls).toHaveLength(1);
    expect(i18n.changeLanguage.mock.calls[0]).toEqual(['fr']);
    */
  });
});
