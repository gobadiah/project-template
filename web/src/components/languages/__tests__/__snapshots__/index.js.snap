// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lang should match snapshot 1`] = `
.emotion-0 {
  width: 100%;
}

<Styled(div)
  main={false}
>
  <Styled(img)
    alt="fr"
    className="emotion-0"
    src="/static/build/329f5c89d4f13938971b9b0d8d35f296.png"
  />
  <Styled(span)>
    FR
  </Styled(span)>
</Styled(div)>
`;

exports[`Languages should match snapshot 1`] = `
.emotion-0 {
  background-color: Transparent;
  border: none;
  cursor: pointer;
  outline: none;
  overflow: hidden;
  padding: 0;
  width: 133px;
  height: 60px;
}

<Styled(div)
  className=""
>
  <Lang
    className="
  width: 133px;
  height: 60px;
"
    lng="fr"
    main={true}
  />
  <Styled(div)>
    <button
      className="emotion-0"
      key="en"
      onClick={[Function]}
    >
      <Lang
        className="
  width: 133px;
  height: 60px;
"
        lng="en"
        main={false}
      />
    </button>
  </Styled(div)>
</Styled(div)>
`;
